pkg/api/plugin_resource.go

```diff
@@ -62,12 +62,7 @@ func (hs *HTTPServer) callPluginResourceWithDataSource(c *contextmodel.ReqContex
 		return
 	}
 
-	var dsURL string
-	if pCtx.DataSourceInstanceSettings != nil {
-		dsURL = pCtx.DataSourceInstanceSettings.URL
-	}
-
-	err = hs.PluginRequestValidator.Validate(dsURL, c.Req)
+	err = hs.PluginRequestValidator.Validate(ds, c.Req)
```

Comment:
to my understanding the `ds` information can come from [the cache](https://github.com/grafana/grafana/blob/8a1bbe7517b94b9e1d2eaf9c13cd674a1036e345/pkg/api/datasources.go#L676) but this change seems fine since the `pCtx.DataSourceInstanceSettings` derive from the [ds](https://github.com/grafana/grafana-plugin-sdk-go/pull/1137#discussion_r1837793951) itself (only difference should be that the secrets in `DataSourceInstanceSettings` are decrypted but we don't use this for any validation